samples/tab-personal-sso-quickstart/ts/src/components/ConsentPopup.tsx

```diff
@@ -2,53 +2,54 @@
 // Licensed under the MIT License.
 
 import React from 'react';
-
-import crypto from 'crypto';
 import * as microsoftTeams from "@microsoft/teams-js";
+import * as msal from "@azure/msal-browser";
 
 /**
- * This component is used to redirect the user to the Azure authorization endpoint from a popup
+ * This component is loaded to grant consent for graph permissions.
  */
 class ConsentPopup extends React.Component {
 
     componentDidMount() {
 
-      // Initialize the Microsoft Teams SDK
-      microsoftTeams.initialize();
-
-      // Get the user context in order to extract the tenant ID
-      microsoftTeams.getContext((context: microsoftTeams.Context) => {
-
-        let tenant = context['tid']; //Tenant ID of the logged in user
-        let client_id = process.env.REACT_APP_AZURE_APP_REGISTRATION_ID; //Client ID of the Azure AD app registration ( may be from different tenant for multitenant apps)
-
-        //Form a query for the Azure implicit grant authorization flow
-        //https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-implicit-grant-flow      
-        let queryParams: any = {
-            tenant: `${tenant}`,
-            client_id: `${client_id}`,
-            response_type: "token", //token_id in other samples is only needed if using open ID
-            scope: "https://graph.microsoft.com/User.Read",
-            redirect_uri: window.location.origin + "/auth-end",
-            nonce: crypto.randomBytes(16).toString('base64')
-        }
-        
-        let url = `https://login.microsoftonline.com/${tenant}/oauth2/v2.0/authorize?`;
-        queryParams = new URLSearchParams(queryParams).toString();
-        let authorizeEndpoint = url + queryParams;
-        
-        //Redirect to the Azure authorization endpoint. When that flow completes, the user will be directed to auth-end
-        //Go to ClosePopup.js
-        window.location.assign(authorizeEndpoint);
-
-      });
-    
-    }    
+        microsoftTeams.app.initialize().then(() => {
+            // Get the tab context, and use the information to navigate to Azure AD login page
+            microsoftTeams.app.getContext().then(async (context) => {
+                var scope = "User.Read email openid profile offline_access Chat.Read Chat.ReadWrite ChatMessage.Send";
+                var loginHint = context.user!.loginHint;
+                //alert(process.env.REACT_APP_MICROSOFT_APP_ID!);
+				
+                const msalConfig: msal.Configuration = {
+                    auth: {
+                        clientId: process.env.REACT_APP_AZURE_APP_REGISTRATION_ID!,
+                        authority: `https://login.microsoftonline.com/${context.user!.tenant!.id}`,
+                        navigateToLoginRequestUrl: false
+                    },
+                    cache: {
+                        cacheLocation: "sessionStorage",
+                    },
+                };
+
+               // alert(JSON.stringify(msalConfig));
```

Comment:
We can remove it and remove extra spaces

Reply:
changed it